src/wallet/test/fuzz/spend.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <key_io.h>
+#include <outputtype.h>
+#include <script/script.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
+#include <validation.h>
+#include <wallet/coincontrol.h>
+#include <wallet/spend.h>
+#include <wallet/test/util.h>
+#include <wallet/wallet.h>
+
+namespace wallet { 
+namespace {
+
+TestChain100Setup* g_setup;
+
+/**
+ * Singleton wallet class ensures that only one 
+ * instance of CWallet is created and reused as required. 
+ * This increases Fuzzing efficiency by reducing the expense
+ * of creating a new `Descriptor Wallet` each time and deletes 
+ * the pointer safely using the destructor.
+ */
+class WalletSingleton {
+public:
+    static WalletSingleton& GetInstance() {
+        static WalletSingleton instance;
+        return instance;
+    }
+
+    CWallet& GetWallet() {
+        if (!wallet) {
+            InitializeWallet();
+        }
+        return *wallet;
+    }
+    CoinsResult& GetCoins() {
+        if (!wallet) {
+            InitializeWallet();
+        }
+        return available_coins;
+    }
+
+private:
+    WalletSingleton() {
+        InitializeWallet();
+    }
+
+    ~WalletSingleton() {
+        wallet.reset();
+        available_coins.coins.clear();
+    }
+
+    WalletSingleton(const WalletSingleton&) = delete;
+    WalletSingleton& operator=(const WalletSingleton&) = delete;
+
+    void InitializeWallet() {
+        auto& node{g_setup->m_node};
+        wallet = CreateSyncedWallet(*node.chain, WITH_LOCK(Assert(node.chainman)->GetMutex(), return node.chainman->ActiveChain()), g_setup->coinbaseKey);

```

Comment:
I think that `CreateSyncedWallet` will call `SetupDescriptorScriptPubKeyMans()` which internally will make a seed.
```cpp
void CWallet::SetupDescriptorScriptPubKeyMans()
{
    AssertLockHeld(cs_wallet);

    if (!IsWalletFlagSet(WALLET_FLAG_EXTERNAL_SIGNER)) {
        // Make a seed
        CKey seed_key;
        seed_key.MakeNewKey(true);
        CPubKey seed = seed_key.GetPubKey();
        assert(seed_key.VerifyPubKey(seed));

        // Get the extended key
        CExtKey master_key;
        master_key.SetSeed(seed_key);

        SetupDescriptorScriptPubKeyMans(master_key);
    }
```

Not sure, but wouldn't it be non-deterministic? A solution would be to have a fuzz version of `CreateSyncedWallet` which calls `SetupDescriptorScriptPubKeyMans` passing a key that we created using buf.